prow/git/git.go

```diff
@@ -138,7 +137,14 @@ func (c *Client) Clone(repo string) (*Repo, error) {
 	base := c.base
 	user, pass := c.getCredentials()
 	if user != "" && pass != "" {
-		base = fmt.Sprintf("https://%s:%s@%s", user, pass, github)
+		gitURL := &url.URL{
+			Scheme: "https",
+			User:   url.UserPassword(user, pass),
+			Host:   c.base,
+			Path:   strings.Join([]string{user, repo}, "/"),
```

Comment:
I don't think this path change needs to be here, the original string pattern was: 

`https://%s:%s@%s`
not
`https://%s:%s@%s/user/repo`